modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/ActionCreationForm.tsx

```diff
@@ -0,0 +1,419 @@
+import RequiredMark from '../components/RequiredMark';
+import openDefaultFailureToast from '../utils/openDefaultFailureToast';
+import openDefaultSuccessToast from '../utils/openDefaultSuccessToast';
+
+const MESSAGE_TYPES = [
+	{
+		label: Liferay.Language.get('info'),
+		value: 'info',
+	},
+	{
+		label: Liferay.Language.get('secondary'),
+		value: 'secondary',
+	},
+	{
+		label: Liferay.Language.get('success'),
+		value: 'success',
+	},
+	{
+		label: Liferay.Language.get('danger'),
+		value: 'danger',
+	},
+	{
+		label: Liferay.Language.get('warning'),
+		value: 'warning',
+	},
+];
+
+const TYPES = [
+	{
+		label: Liferay.Language.get('link'),
+		value: 'link',
+	},
+];
+
+interface IFDSActionCreationFormInterface {
+	fdsView: FDSViewType;
+	getFDSActions: () => void;
+	namespace: string;
+	sections: {
+		ACTIONS: string;
+		NEW_ITEM_ACTION: string;
+	};
+	setActiveSection: (arg: string) => void;
+	spritemap: string;
+}
+
+const ActionCreationForm = ({
+	fdsView,
+	getFDSActions,
+	namespace,
+	sections,
+	setActiveSection,
+	spritemap,
+}: IFDSActionCreationFormInterface) => {
+	const labelFormElementId = `${namespace}Label`;
+	const iconFormElementId = `${namespace}Icon`;
+	const typeFormElementId = `${namespace}Type`;
+	const urlFormElementId = `${namespace}URL`;
+	const confirmationMessageFormElementId = `${namespace}ConfirmationMessage`;
+	const confirmationMessageTypeFormElementId = `${namespace}ConfirmationMessageType`;
+
+	const [availableIconSymbols, setAvailableIconSymbols] = useState<
+		Array<{label: string; value: string}>
+	>([]);
+	const [
+		confirmationMessageTranslations,
+		setConfirmationMessageTranslations,
+	] = useState({});
+	const [labelTranslations, setLabelTranslations] = useState({});
+	const [saveButtonDisabled, setSaveButtonDisabled] = useState(false);
+
+	const [actionData, setActionData] = useState({
+		confirmationMessage: '',
+		iconSymbol: '',
+		label: '',
+		type: 'link',
+		url: '',
+	});
+
+	const saveFDSAction = async () => {
+		setSaveButtonDisabled(true);
+
+		const {confirmationMessage, iconSymbol, type, url} = actionData;
+
+		const body = {
+			[OBJECT_RELATIONSHIP.FDS_VIEW_FDS_ACTION_ID]: fdsView.id,
+			icon: iconSymbol,
+			type,
+			url,
+		} as any;
+
+		if (Liferay.FeatureFlags['LPS-172017']) {
+			body.confirmationMessage_i18n = confirmationMessageTranslations;
+			body.label_i18n = labelTranslations;
+		}
+		else {
+			body.confirmationMessage = confirmationMessage;
+			body.label = labelTranslations;
+		}
+
+		const response = await fetch(API_URL.FDS_ACTIONS, {
+			body: JSON.stringify(body),
+			headers: {
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+			},
+			method: 'POST',
+		});
+
+		if (!response.ok) {
+			setSaveButtonDisabled(false);
+
+			openDefaultFailureToast();
+
+			return;
+		}
+
+		await response.json();
+
+		setSaveButtonDisabled(false);
+
+		openDefaultSuccessToast();
+
+		setActiveSection(sections.ACTIONS);
+
+		getFDSActions();
```

Comment:
This is a little odd. Maybe it should be called something like `onSave`? Otherwise it makes me think we are trying to "get" something but not doing anything with it.

Reply:
`loadFDSActions` maybe? We are _getting_ them, from backend, but we are also applying them in when fetch resolves. `reload` probably sounds best in this context, but we're also doing this on first render.

Reply:
Changed to `loadFDSActions`